### PR TITLE
EOS-23581: Prevent `/etc/sudoers.d/sspl` file deletion

### PR DESCRIPTION
**Pull request by sumedhak27**
Signed-off-by: Sumedh A. Kulkarni <sumedh.a.kulkarni@seagate.com>

## Problem Statement:
Puppet-agent is deleting the /etc/sudoers.d/sspl file since the cmd to write noop = true has some issues. correction regarding the same is needed.

## Coding:

* [x] Did you follow coding standard and verified it with flake8? 

* [x] Did your address all Codacy issues?

## Testing:

* [x] Unittest updated and all unittests are passing?

* [x] Sanity/Self tests are updated (if applicable)?

* [x] Manual testing done covering happy path and non-happy path?

## Integration:

* [ ] If there is any interface change, did you communicate it to other components?
* [ ] If there is any interface change, other componenent gate gatekeepers are ready to accept the change?
* [ ] Did you complete deployment test if applicable?

## PR checklist:
* [x] Did you add Jira number to the PR?

* [x] DCO and cla-signed?

### CODE REVIEW

File: low-level/framework/sspl_reinit [hunk 1]
```diff
@@ -60,8 +60,13 @@ chmod 440 /etc/sudoers.d/sspl
...
 [ -f $PUPPET_CONF ] && {
-    PUPPET_NOOP="noop = true"
-    grep -q "$PUPPET_NOOP" $PUPPET_CONF || echo "$PUPPET_NOOP" >> $PUPPET_CONF
+    grep -q "noop" $PUPPET_CONF
+    if [ $? -eq 0 ]
+    then
+        sed -i -E 's/noop\ *=\ *false/noop = true/g' $PUPPET_CONF
+    else
+        echo "noop = true" >> $PUPPET_CONF
+    fi
```
Comment (sumedhak27) [inline low-level/framework/sspl_reinit:69]:
```shell
PUPPET_CONF="/etc/puppetlabs/puppet/puppet.conf"
[ -f $PUPPET_CONF ] && {
    grep -q "noop = false" $PUPPET_CONF && sed -i "s/noop = false/noop = true/" $PUPPET_CONF
    grep -q "noop = true" $PUPPET_CONF || echo "noop = true" >> $PUPPET_CONF
}
```
@mariyappanp is suggesting this simpler way considering modern shell script.
But I wanted to keep readability and consider the cases with multiple spaces before & after the '=' sign.
Everyone, please suggest which option we should go with.

File: low-level/framework/sspl_reinit [hunk 2]
```diff
@@ -60,8 +60,13 @@ chmod 440 /etc/sudoers.d/sspl
...
 [ -f $PUPPET_CONF ] && {
-    PUPPET_NOOP="noop = true"
-    grep -q "$PUPPET_NOOP" $PUPPET_CONF || echo "$PUPPET_NOOP" >> $PUPPET_CONF
+    grep -q "noop\s*=.*" $PUPPET_CONF
+    if [ $? -eq 0 ]
+    then
+        sed -i -E 's/noop\ *=\ *false/noop = true/g' $PUPPET_CONF
+    else
+        echo "noop = true" >> $PUPPET_CONF
+    fi
```
Comment (sumedhak27) [inline low-level/framework/sspl_reinit:69]:
```shell
PUPPET_CONF="/etc/puppetlabs/puppet/puppet.conf"
[ -f $PUPPET_CONF ] && {
    grep -q "noop = false" $PUPPET_CONF && sed -i "s/noop = false/noop = true/" $PUPPET_CONF
    grep -q "noop = true" $PUPPET_CONF || echo "noop = true" >> $PUPPET_CONF
}
```
@mariyappanp is suggesting this simpler way considering modern shell script.
But I wanted to keep readability and consider the cases with multiple spaces before & after the '=' sign.
Everyone, please suggest which option we should go with.

**Review by mariyappanp — APPROVED** [review]

File: low-level/framework/sspl_reinit [hunk 3]
```diff
@@ -59,10 +59,7 @@ chmod 440 /etc/sudoers.d/sspl
 # Set noop = true in puppet conf file to avoid the sspl auth file getting
 # removed by puppet-agent (Ref: LRL-495)
 PUPPET_CONF="/etc/puppetlabs/puppet/puppet.conf"
-[ -f $PUPPET_CONF ] && {
-    PUPPET_NOOP="noop = true"
-    grep -q "$PUPPET_NOOP" $PUPPET_CONF || echo "$PUPPET_NOOP" >> $PUPPET_CONF
-}
+[ -f $PUPPET_CONF ] && conf "ini://$PUPPET_CONF" set "agent>noop=true"
```
Comment (mariyappanp) [inline low-level/framework/sspl_reinit:62]: This script is being called during sspl post_install. At that time, consider only cortx-utils post_install is just completed. 
In this case, does cortx-utils cli become usable once post_install stage completed for cortx-utils? If not, this approach can not work. Please make sure verified this on a clean setup. @sumedhak27 

cortx-utils conf cli will always be

**Review by mariyappanp — CHANGES_REQUESTED** [review]
Please confirm this.

File: low-level/framework/sspl_reinit (hunk 3 above)
Comment (mariyappanp) [reply low-level/framework/sspl_reinit:62]: Sumedh confirmed this test. Hence approving.

**Review by mariyappanp — APPROVED** [review]

**Review by shriya-deshmukh — APPROVED** [review]

**Review by cdeshmukh — APPROVED** [review]

**Comment by cdeshmukh** [comment]
Vm deployment success http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Cortx-Deployment/job/SSPL/572/

**Pull request merged**